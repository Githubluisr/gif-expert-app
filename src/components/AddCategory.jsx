import { useState } from "react";
import  PropTypes  from 'prop-types' 

export const AddCategory = ( { newCategory } ) => {
	const [inputValue, setInputValue] = useState("");

	const onInputChange = (e) => {
		setInputValue(e.target.value);
	};

  const onSubmit = (e) => {
    e.preventDefault();

    if (inputValue.length <= 0 ) return;

    newCategory(inputValue);
    setInputValue('');
  }

	return (
		<form onSubmit={onSubmit} aria-label='form'>
			<input
				type="text"
				placeholder="Nueva categoria"
				value={inputValue}
				onChange={onInputChange}
			></input>
		</form>
	);
};

AddCategory.propTypes = {
	newCategory: PropTypes.func.isRequired
}
