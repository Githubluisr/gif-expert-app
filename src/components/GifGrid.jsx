import { GifItem } from './GifItem'
import { useFetchGifs } from "./hooks/useFetchGifs";
import PropTypes from 'prop-types'

export const GifGrid = ( {category} ) => {

  const {images, isLoading} = useFetchGifs(category); //Customm hook

  return (
    <>
      <div className='clear-fix'></div>
      <h3>{category}</h3>
      { isLoading && <h2>Loading...</h2> }
      {
        images.map((images) => (
          <GifItem key={images.id} {...images} />
        ))
      }
      
      
    </>
    
  )
}


GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}