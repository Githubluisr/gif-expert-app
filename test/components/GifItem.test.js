import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem';

describe('Test on GifItem components', () => {
  const category = 'Han%20Solo' //tuve que buscarlo asi porque al regresarlo en la prueba le adiciona &20

  const images = {
    id:1,
    title: 'Title Test', 
    url: `https://api.giphy.com/v1/gifs/search?api_key=A97WbUNE1Ed9tUwxzRWgHyKncplinPrf&q=${category}&limit=10`
  }

  test('Must match snapshot', () => {
   
    // const title = "Saitama";
    // const url=`https://api.giphy.com/v1/gifs/search?api_key=A97WbUNE1Ed9tUwxzRWgHyKncplinPrf&q=${category}&limit=10`
    const { container } = render(<GifItem key={1}{ ...images }/>)

    expect(container).toMatchSnapshot();
  });

  test('Alt and url must be correct', () => {


    render(<GifItem key={1}{ ...images }/>)
    //screen.debug();

    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(images.url);
    expect(alt).toBe(images.title);
    
  })


  test('Mostrar title en la pagina', ()=> {

    render(<GifItem key={1}{ ...images }/>)

    expect(screen.getByText(images.title));
  })

});