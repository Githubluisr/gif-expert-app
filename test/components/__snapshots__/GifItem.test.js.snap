// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifItem components Must match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title Test"
      src="https://api.giphy.com/v1/gifs/search?api_key=A97WbUNE1Ed9tUwxzRWgHyKncplinPrf&q=Han%20Solo&limit=10"
    />
    <p>
      Title Test
    </p>
  </div>
</div>
`;
